Framework/PythonInterface/plugins/algorithms/NOMADMedianDetectorTest.py

```diff
@@ -0,0 +1,146 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+# package imports
+from mantid.api import (AlgorithmFactory, FileAction, FileProperty, MatrixWorkspace, PropertyMode, PythonAlgorithm,
+                        WorkspaceProperty)
+from mantid.kernel import Direction
+from mantid.simpleapi import (DeleteWorkspaces, Divide, ExtractMask, LoadEmptyInstrument, MaskDetectors, mtd, SaveMask,
+                              SolidAngle)
+
+from mantid.utils.nomad.diagnostics import _NOMADMedianDetectorTest
+
+# third-party imports
+import numpy as np
+
+# standard imports
+import os
+
+
+class NOMADMedianDetectorTest(PythonAlgorithm, _NOMADMedianDetectorTest):
```

Comment:
Is it possible to move the relevant content directly to this algorithm?
Hiding implementation in a different file will cause more headache down the line.